### PR TITLE
Update SSH PAM tutorial to include new features.

### PR DESCRIPTION
**Pull request by yashtewari**
Tutorial now uses display and pull policies to request authorization context.
Separate policies for sudo and SSH have been merged into one for simplicity.

Signed-off-by: Yash Tewari <yashtewari1996@gmail.com>

### CODE REVIEW

File: docs/book/ssh-and-sudo-authorization.md [hunk 1]
```diff
@@ -256,85 +263,137 @@ able to login.
...
+for both the SSH and `sudo` authorization cycles.
+This happens because the *display* policy is shared by the PAM configurations of SSH and `sudo`.
+In production, it is more practical to use separate policy packages for each PAM configuration.
 
-Lastly, remove the policy from OPA and confirm the user's original rights are restored.
 
-```shell
-curl -i -X DELETE localhost:8181/v1/policies/sudo_authz_special
-```
+For SSH, the value entered does not matter.
```
Comment (arcurtis) [inline docs/book/ssh-and-sudo-authorization.md]: Would rewrite: "The SSH PAM policy does not rely on the input ticket number, so the value does not affect the authorization decision"

File: docs/book/ssh-and-sudo-authorization.md [hunk 2]
```diff
@@ -256,85 +263,137 @@ able to login.
...
+  "tickets": {}
+}'
 ```
 
+You will find that running `sudo ls /` as the `frontend-dev` user is disallowed again.
+
+It is possible to configure the *display* policy to only make the PAM module prompt for the
+elevation ticket when our mock API has a non-empty `tickets` object. So when there are no
+elevated users, there will be no prompt for a ticket. Doing this has been left as an
+exercise for the reader.
```
Comment (arcurtis) [inline docs/book/ssh-and-sudo-authorization.md]: Somewhat a personal pet peeve, but "left as an exercise for the reader" is really annoying! Either explain how to do it (doesn't have to have the policy, but at least an explanation of what one would do) or cut that sentence entirely.

File: docs/book/ssh-and-sudo-authorization.md [hunk 3]
```diff
@@ -256,85 +263,137 @@ able to login.
...
  You learned a number of things about SSH with OPA:
 
-* OPA gives you fine-grained policy control over SSH and sudo on any host where you've
-  installed OPA and its PAM module.
-* You write allow/deny policies to control who has access to what.
-* You can import external data into OPA and write policies that depend on.
-  that data.
+* OPA gives you fine-grained policy control over SSH, `sudo`, and any other application that uses PAM,
+  on any host where you've
+  installed OPA and its PAM module. Although this tutorial used the some of the same policies for both
```
Comment (arcurtis) [inline docs/book/ssh-and-sudo-authorization.md]: Would cut " on any host where you've installed OPA and its PAM module." as that is clumsy and should be clear to the reader at this point

File: docs/book/ssh-and-sudo-authorization.md [hunk 4]
```diff
@@ -256,85 +263,137 @@ able to login.
...
+  SSH and sudo, you should use separate, fine-grained policies for each application that supports PAM.
+* You write allow/deny policies to control who has access to what, based on authorization context generated
```
Comment (arcurtis) [inline docs/book/ssh-and-sudo-authorization.md]: Would rewrite: "You write allow/deny policies to control who has access to what, based on authorization context generated using steps defined in other policies." to 

"Writing allow/deny policies to control who has access to what using context from the user, servers, and a ticketing system."

File: docs/book/ssh-and-sudo-authorization.md (hunk 1 above)
Comment (yashtewari) [reply docs/book/ssh-and-sudo-authorization.md]: Done.

File: docs/book/ssh-and-sudo-authorization.md (hunk 2 above)
Comment (yashtewari) [reply docs/book/ssh-and-sudo-authorization.md]: Removed, and added a hint.

File: docs/book/ssh-and-sudo-authorization.md (hunk 3 above)
Comment (yashtewari) [reply docs/book/ssh-and-sudo-authorization.md]: Done.

File: docs/book/ssh-and-sudo-authorization.md (hunk 4 above)
Comment (yashtewari) [reply docs/book/ssh-and-sudo-authorization.md]: Done, except for the ticketing system bit which is generalized in the next bullet point.

File: docs/book/ssh-and-sudo-authorization.md (hunk 2 above)
Comment (arcurtis) [reply docs/book/ssh-and-sudo-authorization.md]: Thanks :)

File: docs/book/ssh-and-sudo-authorization.md (hunk 4 above)
Comment (arcurtis) [reply docs/book/ssh-and-sudo-authorization.md]: Awesome, even better

**Pull request merged**